src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Windows.cs

```diff
@@ -446,8 +446,8 @@ private static bool CanUseScatterGatherWindowsAPIs(SafeFileHandle handle)
         // The pinned MemoryHandles and the pointer to the segments must be cleaned-up
         // with the CleanupScatterGatherBuffers method.
         private static unsafe bool TryPrepareScatterGatherBuffers<T, THandler>(IReadOnlyList<T> buffers,
-            THandler handler, [NotNullWhen(true)] out MemoryHandle[]? handlesToDispose, out IntPtr segmentsPtr, out int totalBytes)
-            where THandler : struct, IMemoryHandler<T>
```

Comment:
It is good to enforce specialized binding to the interface, but I removed it because there was no apparent reason (and didn't see it in the `SslStream` code I linked).

Reply:
Constraining generic arguments by `structs` can give hint to some of the AOT compilers that they do not need to bother to create code shared over reference instantiations.

Reply:
Both T and THandler can be constrained as `structs` here.

Reply:
Done.

Reply:
> didn't see it in the SslStream code I linked

Maybe this can be done in the SslStream code too?

Reply:
There are 21 places `where TAdapter : IReadWriteAdapter` is being used; it would go beyond this PR's scope, better for another time.

BTW can't the AOT compilers be smarter and detect that the shared instantiation will not be used? Or is it done intentionally (my guess would be for something like a fallback for reflection)?

Reply:
> Or is it done intentionally (my guess would be for something like a fallback for reflection)?

Correct. I believe that some of the AOT compilers that we have do this already ... but it is hard to do in general case due to reflection.